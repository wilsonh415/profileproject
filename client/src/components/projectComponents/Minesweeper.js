import React from 'react';
import { Card, Nav, Button } from 'react-bootstrap';
import minesweeper from '../../images/minesweeperimg.png';

class Minesweeper extends React.Component {

    render() {
        return (
            <div className="msCardClass">
                <Card>
                    <Card.Img variant="top" src={minesweeper} width={286} height={180}/>
                    <Card.Body>
                        <Card.Title><b>Minesweeper</b></Card.Title>
                        <b><hr/></b>
                        <Card.Text>
                            <b>Rules</b>: Clear every tile without selecting
                            a mine and you win. Each "safe" tile will have a
                            number indicating the number of mines around it. Choose carefully. Good luck.
                        </Card.Text>
                        <b><hr/></b>
                        <Nav.Link href='/Minesweeper'>
                            <Button variant="primary">Play now!</Button>
                        </Nav.Link>
                    </Card.Body>
                </Card>
            </div>
        )
    }
}

export default Minesweeper;