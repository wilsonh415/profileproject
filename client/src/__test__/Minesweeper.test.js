import React from 'react';
import {shallow} from 'enzyme';
import {configure} from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Minesweeper from '../components/projectComponents/Minesweeper';

configure({adapter: new Adapter()});

test('Minesweeper component rendererd properly', () => {
    const component = renderer.create(<Minesweeper></Minesweeper>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});