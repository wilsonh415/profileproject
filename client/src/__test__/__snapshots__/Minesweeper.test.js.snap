// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minesweeper component rendererd properly 1`] = `
<div
  className="msCardClass"
>
  <div
    className="card"
  >
    <img
      className="card-img-top"
      height={180}
      src="minesweeperimg.png"
      width={286}
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        <b>
          Minesweeper
        </b>
      </div>
      <b>
        <hr />
      </b>
      <p
        className="card-text"
      >
        <b>
          Rules
        </b>
        : Clear every safe tile without selecting a mine and you win. Each safe tile will have a number indicating the number of mines around it. Choose carefully. Good luck.
      </p>
      <b>
        <hr />
      </b>
      <a
        className="nav-link"
        href="/Minesweeper"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Play now!
        </button>
      </a>
    </div>
  </div>
</div>
`;
